govcd/guestproperties.go

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+// setGuestProperties is a shared function for
+func setGuestProperties(client *Client, href string, properties *types.ProductSectionList) error {
+	if href == "" {
+		return fmt.Errorf("href cannot be empty to set guest properties")
+	}
+
+	properties.Xmlns = types.XMLNamespaceVCloud
+	properties.Ovf = types.XMLNamespaceOVF
+
+	task, err := client.ExecuteTaskRequest(href+"/productSections", http.MethodPut,
```

Comment:
kind'a lost here type producSections, but variable name is properties

Reply:
Yeah. The endpoint is such therefore struct name is as it is, but in fact we are dealing with "guest properties" from vCD perspective. It matches function names and UI.